OpenRA.Mods.Common/Widgets/ModelWidget.cs

```diff
@@ -0,0 +1,220 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2020 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using OpenRA.Graphics;
+using OpenRA.Mods.Common.Graphics;
+using OpenRA.Widgets;
+
+namespace OpenRA.Mods.Common.Widgets
+{
+	public class ModelWidget : Widget
+	{
+		public string Palette = "terrain";
+		public string PlayerPalette = "player";
+		public string NormalsPalette = "normals";
+		public string ShadowPalette = "shadow";
+		public float Scale = 12f;
+		public int LightPitch = 142;
+		public int LightYaw = 682;
+		public float[] LightAmbientColor = new float[] { 0.6f, 0.6f, 0.6f };
+		public float[] LightDiffuseColor = new float[] { 0.4f, 0.4f, 0.4f };
+		public WRot Rotation = new WRot(new WAngle(0), new WAngle(0), new WAngle(0));
+		public WAngle CameraAngle = WAngle.FromDegrees(40);
+
+		public Func<string> GetPalette;
+		public Func<string> GetPlayerPalette;
+		public Func<string> GetNormalsPalette;
+		public Func<string> GetShadowPalette;
+		public Func<float[]> GetLightAmbientColor;
+		public Func<float[]> GetLightDiffuseColor;
+		public Func<float> GetScale;
+		public Func<int> GetLightPitch;
+		public Func<int> GetLightYaw;
+		public Func<IModel> GetVoxel;
+		public Func<WRot> GetRotation;
+		public Func<WAngle> GetCameraAngle;
+		public int2 PreviewOffset { get; private set; }
```

Comment:
These aren't used outside of this widget, and don't make much sense for widgets at all.